ray-curator/ray_curator/stages/deduplication/io_utils.py

```diff
@@ -0,0 +1,57 @@
+from collections.abc import Callable
+
+# TODO: Should this be a safe import?
+import cudf
+import numpy as np
+import ray
+
+from ray_curator.stages.deduplication.id_generator import CURATOR_DEDUP_ID_STR, IdGenerator
+from ray_curator.utils.file_utils import get_fs
+
+
+class DeduplicationIO:
+    def __init__(
+        self,
+        id_generator: "IdGenerator | None",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.id_generator = id_generator
+
+    def read_jsonl(
+        self, filepath: str | list[str], columns: list[str] | None = None, assign_id: bool = False, **kwargs
+    ) -> "cudf.DataFrame":
+        df = cudf.read_json(filepath, lines=True, **kwargs)
+        if columns is not None:
+            df = df[columns]
+        return self.assign_id(filepath, df) if assign_id and self.id_generator else df
+
+    def read_parquet(self, filepath: str | list[str], assign_id: bool = False, **kwargs) -> "cudf.DataFrame":
+        read_kwargs = kwargs.copy()
+        read_kwargs["allow_mismatched_pq_schemas"] = True
+        df = cudf.read_parquet(filepath, **read_kwargs)
+        return self.assign_id(filepath, df) if assign_id and self.id_generator else df
+
+    def write_parquet(self, df: "cudf.DataFrame", filepath: str, **kwargs) -> None:
+        fs = get_fs(filepath, storage_options=kwargs.get("storage_options", {}))
+        # TODO: Add overwrite behavior here
+        fs.makedirs(fs._parent(filepath), exist_ok=True)
```

Comment:
Previous behavior was to overwrite if directory already existed. Might want to keep the same here.